front_end/panels/ai_chat/ui/SettingsDialog.ts

```
@@ -442,8 +532,17 @@ export class SettingsDialog {
     
-    // Get the current provider from localStorage
-    const currentProvider = localStorage.getItem(PROVIDER_SELECTION_KEY) || 'openai';
+    // Determine current provider: prefer the provider for the currently selected model if available
+    const storedProvider = localStorage.getItem(PROVIDER_SELECTION_KEY) || 'openai';
+    let currentProvider = storedProvider as 'openai' | 'litellm' | 'groq' | 'openrouter';
+    try {
+      const selectedModelOption = modelOptions.find(opt => opt.value === selectedModel);
+      if (selectedModelOption && selectedModelOption.type) {
+        currentProvider = selectedModelOption.type as 'openai' | 'litellm' | 'groq' | 'openrouter';
+      }
+    } catch (e) {
```

ESLint

[error] 'e' is defined but never used.

(@typescript-eslint/no-unused-vars)